torchgen/gen_functionalization_type.py

```diff
@@ -618,7 +631,9 @@ def emit_inplace_functionalization_body(
       }}
       {unwrap_tensor_args_str}
       if (!({check_all_mutated_args_are_functional})) {{
-        if (({check_any_non_mutated_args_are_functional})) {{
+        // We want to disable this check if there are any XLA tensors.
+        // cpu_tensor.copy_(xla_tensor) is valid code.
+        if ((!{check_any_non_mutated_tensors_are_xla}) && ({check_any_non_mutated_args_are_functional})) {{
```

Comment:
Trying this fix out locally, this line generates:
```
if ((!false || src.device().type() == c10::DeviceType::XLA) && (false || at::functionalization::impl::isFunctionalTensor(src))) {
```
which seems not what we want to do, since the first part `(!false || src.device().type() == c10::DeviceType::XLA)` equates to `true` every time. 

I've updated this locally to:
```
if (!({check_any_non_mutated_tensors_are_xla}) && ({check_any_non_mutated_args_are_functional})) {{
```
which generates:
```
if (!(false || src.device().type() == c10::DeviceType::XLA) && (false || at::functionalization::impl::isFunctionalTensor(src))) {
```
and I can see that this fixes XLA's tests with `copy_`. 

Just wanted to check with you, @bdhirsh, if this fix is okay. Thanks!

Reply:
ah you're right, I messed up the parens. Thanks wonjoo, I'll update!

Reply:
No worries! And no need to update this PR if you're not planning to merge this one. I can just update the original PR to include this change https://github.com/pytorch/pytorch/pull/88787/. Thanks!